microprofile/WFLY-12342_server_readiness_probes.adoc

```diff
@@ -0,0 +1,95 @@
+= [WFLY-12342] Integrate server probes in MP Health readiness check
+:author:            Jeff Mesnil
+:email:             jmesnil@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+:keywords:          observability,microprofile,health,openshift
+
+== Overview
+
+MicroProfile Health 2.0 introduced a readiness health check.
+WFLY-12228 added support for MicroProfile Health 2.0 without enhancing the readiness health check.
+
+This means that the readiness probe always return UP by default (as long as the microprofile-health-smallrye is installed).
+
+This RFE enhances it with 3 probes (that are part of the https://github.com/jboss-container-images/jboss-eap-modules/blob/251f422c97e1bcd3625f57295bc79973193482a6/os-eap-probes/2.0/added/probes/probe/eap/dmr.py#L23[EAP readiness probe script]):
+
+* `server-status` - returns UP when the server-state is `running`
+* `boot-errors` - return UP if there are no boot-errors
+* `deployment-status` - returns UP if status for all deployments is OK
+
+Once this RFE is done, we could simplify the EAP readiness probe script by replacing its various DMR checks by a single check to `/subsystem=microprofile-health-smallrye:check-ready` (or a HTTP call to `http://:9990/health/ready`).
```

Comment:
I can't see `/subsystem=microprofile-health-smallrye:check-ready` in wf20

Reply:
that's weird, I confirm it's already in WildFly 20.0.0.Final: https://docs.wildfly.org/20/wildscribe/subsystem/microprofile-health-smallrye/index.html#op-check-ready

```
[standalone@localhost:9990 /] /subsystem=microprofile-health-smallrye:check-ready
{
    "outcome" => "success",
    "result" => {
        "status" => "UP",
        "checks" => []
    }
}
```

Reply:
I confirm I can see as well.

I was using embeded server. Unless this is expected I will fill a jira for that

```
./jboss-cli.sh 
You are disconnected at the moment. Type 'connect' to connect to the server or 'help' for the list of supported commands.
[disconnected /] embed-server 
[standalone@embedded /] /subsystem=microprofile-health-smallrye:check-ready
{
    "outcome" => "failed",
    "failure-description" => "WFLYCTL0031: No operation named 'check-ready' exists at address [(\"subsystem\" => \"microprofile-health-smallrye\")]",
    "rolled-back" => true
}
```